src/extension.ts

```diff
@@ -100,53 +98,51 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
         refreshEvent: ext.emitters.refreshWorkspaceTree.event,
     });
 
-    const v2ApiFactory = () => {
-        if (v2Api === undefined) {
-            v2Api = {
-                apiVersion: '2.0.0',
-                resources: createAzureResourcesHostApi(
-                    azureResourceProviderManager,
-                    azureResourceBranchDataProviderManager,
-                    azureResourceTreeDataProvider,
-                    workspaceResourceProviderManager,
-                    workspaceResourceBranchDataProviderManager,
-                    workspaceResourceTreeDataProvider,
-                ),
-                activity: {
-                    registerActivity
+    const v2ApiFactory: AzureExtensionApiFactory<AzureResourcesApiInternal> = {
+        apiVersion: '2.0.0',
+        createApi: (options: GetApiOptions) => {
+            return createWrappedAzureResourcesExtensionApi(
```

Comment:
Is this pattern going to create a new API object for each extension--thus, separate lists of registrations, etc.?

Reply:
No, it just wraps the same API instance with new telemetry handling each time.